audit/entry_formatter.go

```diff
@@ -82,15 +83,21 @@ func (f *EntryFormatter) Process(ctx context.Context, e *eventlogger.Event) (*ev
 		return nil, fmt.Errorf("%s: cannot parse event payload: %w", op, event.ErrInvalidParameter)
 	}
 
-	var result []byte
+	if a.Data == nil {
+		return nil, fmt.Errorf("%s: cannot audit event (%s) with no data: %w", op, a.Subtype, event.ErrInvalidParameter)
+	}
+
+	// Take a copy of the event data before we modify anything.
 	data := new(logical.LogInput)
-	headers := make(map[string][]string)
+	dataCopy, err := copystructure.Copy(a.Data)
+	if err != nil {
+		return nil, fmt.Errorf("%s: unable to copy audit event data: %w", op, err)
+	}
+	data = dataCopy.(*logical.LogInput)
 
-	if a.Data != nil {
-		*data = *a.Data
-		if a.Data.Request != nil && a.Data.Request.Headers != nil {
-			headers = a.Data.Request.Headers
-		}
+	headers := make(map[string][]string)
```

Comment:
Same point here, this allocates memory for an empty map of string keys to string slices. But then the variable is reassigned to `data.Request.Headers`. Either change this line to a variable declaration like `var headers map[string][]string` or change line 100 below to actually copy the elements from the `data.Request.Headers` map into the **headers** map.